src/main/kotlin/blackjack/domain/Player.kt

```diff
@@ -0,0 +1,55 @@
+package blackjack.domain
+
+class Player(val name: String, cards: PlayerCards) {
+    var cards: PlayerCards = cards
+        private set
+
+    var state: States = States.HIT
+        get() {
+            if (field == States.STAY || field == States.BLACK_JACK) {
+                return field
+            }
+
+            return findStateByScore(cards.score)
+        }
+
+    val isPlaying: Boolean
+        get() {
+            return state == States.HIT
+        }
+
+    init {
+        if (cards.score == Game.BLACK_JACK_SCORE) {
+            state = States.BLACK_JACK
+        }
+    }
+
+    constructor(name: String, cards: Set<Card>) : this(name, PlayerCards(cards))
+    constructor(name: String, vararg cards: Card) : this(name, PlayerCards(cards.toSet()))
+
+    private fun findStateByScore(score: Int): States {
+        if (score < Game.BLACK_JACK_SCORE) {
+            return States.HIT
+        } else if (score > Game.BLACK_JACK_SCORE) {
+            return States.BUST
+        }
+
+        return States.STAY
+    }
```

Comment:
when expression을 활용하면 조금 더 가독성 있는 코드를 작성할 수 있습니다.
```suggestion
    private fun findStateByScore(score: Int): States = when {
        score < Game.BLACK_JACK_SCORE ->
            States.HIT
        score > Game.BLACK_JACK_SCORE ->
            States.BUST
        else ->
            States.STAY
    }
```